### PR TITLE
New dad joke function

### PR DESCRIPTION
**Pull request by thekanbo**
Stolen from @hlmtre

### CODE REVIEW

File: handlers/dad.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const _       = require('lodash');
```
Comment (skeltonath) [inline handlers/dad.js]: don't use lodash anywhere in here, byebye

File: handlers/dad.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+const request = require('request');
```
Comment (skeltonath) [inline handlers/dad.js]: don't use request, byebye

File: handlers/dad.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+const rp      = require('request-promise');
+const cheerio = require('cheerio');
```
Comment (skeltonath) [inline handlers/dad.js]: don't use cheerio, byebye

File: handlers/dad.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+const format  = require('format');
```
Comment (skeltonath) [inline handlers/dad.js]: don't use format, byebye

File: handlers/dad.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+let options = {
+    uri: DAD_URL,
+    json: true
+  };
+
+  rp.get(options)
+    .then(joke => {
+      channel.send(joke.joke);
+    });
```
Comment (skeltonath) [inline handlers/dad.js]: log the error case?

File: handlers/dad.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+ */
+module.exports = {
+  name: 'dad',
+  command: 'dad',
+  handler: dad
+};
+
+const DAD_URL = 'https://icanhazdadjoke.com/';
+
+async function dad(channel, message, params) {
```
Comment (skeltonath) [inline handlers/dad.js]: doesn't need to be async

File: handlers/dad.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+const cheerio = require('cheerio');
+const format  = require('format');
+const log4js  = require('log4js');
+
+const LOG = log4js.getLogger('dad');
+
+/**
+ * This module takes a string and inserts into
+ * a movie title and description.
+ */
```
Comment (skeltonath) [inline handlers/dad.js]: update comment

**Comment by thekanbo** [comment]
REVIEW PLZ

File: handlers/dad.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  name: 'dad',
+  command: 'dad',
+  handler: dad
+};
+
+const DAD_API = 'https://icanhazdadjoke.com/';
+
+function dad(channel, message, params) {
+
+let options = {
```
Comment (skeltonath) [inline handlers/dad.js]: Indentation!!!!!!

File: handlers/dad.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+    uri: DAD_API,
+    json: true
+  };
+
+  rp.get(options)
+    .then(joke => {
+      if(joke.joke){
```
Comment (skeltonath) [inline handlers/dad.js]: Space after `if` and between `){`

File: handlers/dad.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+        channel.send(joke.joke);
+      } else {
+        LOG.error(joke);
+        channel.send('Error receiving joke from Dad Central Station');
+      }
+    })
+    .catch(err =>{
```
Comment (skeltonath) [inline handlers/dad.js]: Space after `=>`

**Comment by skeltonath** [comment]
Minor comments. You can merge after fixing

**Pull request merged**